### ISSUE TITLE
MySQL driver uses double quotes for updates, but backtick quotes for select

### ISSUE THREAD

**Issue by chadjaros** [slick/slick#1421]
This breaks down when trying to unit test with H2 in mysql mode. H2 has its own set of WTFs with mysql because it upper cases tables and schemas. The mixing of double quotes and backticks makes it impossible to find a configuration which works for unit testing on H2.

I would suggest using backticks everywhere.

**Comment by chadjaros** [comment]
Findings:

INSERT -> double quotes
UPDATE -> double quotes

SELECT -> backtick
INSERT ON DUPLICATE KEY UPDATE -> backtick
DELETE -> backtick

**Comment by nafg** [comment]
You're using the mysql driver with h2?

**Comment by chadjaros** [comment]
Using the MySQL driver for production. Using the h2 driver for unit tests with MODE=MYSQL

    jdbc:h2:mem:junit;MODE=MYSQL;DATABASE_TO_UPPER=false

**Comment by chadjaros** [comment]
I see your point... it's likely not the driver that's causing the issue but other sql generator stuff

**Comment by chadjaros** [comment]
Correction - It seems like the problem is in the H2 driver

**Comment by Sumith1896** [comment]
Can I take this issue and fix it?

**Comment by chadjaros** [comment]
Yes pls. I tried browsing through the source code to find the root, but I couldn't find it.

**Comment by elacin** [comment]
This might be the same as #166 . To work around that I've used something like this in the past:
```scala
object MyH2Driver extends H2Driver {
  override def quoteIdentifier(id: String) = id
}
```

**Comment by chadjaros** [comment]
That seems related, but the fact that there are two different quote characters being used seems like a bug outside of the sql interpolation issue. 

I'm not sure why it would be necessary to uppercase identifiers in Slick's interpolation, I find that fairly sketchy as well. I filed this item against H2: https://groups.google.com/forum/#!topic/h2-database/iR6oXzJ4syA, but if Slick is causing the uppercasing for backtick'd ids then it should be fixed here.

**Comment by chadjaros** [comment]
My workaround was to use https://github.com/vorburger/MariaDB4j instead of H2 for unit testing. That's probably a better solution anyways since our production server is MySQL. Everything is a bit slower, but totally functional now.

**Comment by joesan** [comment]
Is this issue fixed? I'm facing this issue:

https://stackoverflow.com/questions/48955518/slick-3-0-error-when-selecting-from-h2-database

Here is what I'm using:

  "mysql" % "mysql-connector-java" % "5.1.26",
  "com.h2database" % "h2" % "1.4.186",

**Comment by hvesalai** [comment]
@Sumith1896 did you have a PR for this?

**Comment by Sumith1896** [comment]
@hvesalai No I didn't have one, sorry about that

**Comment by ysono** [comment]
Adding `H2Profile` to our jdbc profile fixed it for us:

```
trait Datastore extends JdbcProfile {...}

class DatastoreImpl @Inject() (database: DatabaseDef) extends Datastore with MySQLProfile {...}

class DatastoreImplForTest (database: DatabaseDef) extends Datastore with H2Profile {...}
```

**Comment by d6y** [comment]
I can't be sure, but this issue looks like mixing (e.g.) MySQL Slick profile with an H2 database. @ysono's fix supports my assumption.

If that's the case, there's no code to fix. But it does suggest we need to document that a profile and database need to match.  

If that is the case...

- We've written [something on this](http://books.underscore.io/essential-slick/essential-slick-3.html#abstracting-over-databases); but 

- perhaps I could find somewhere in the Slick manual to flag up the profile choice (if it's not already clear)
